db/view/view_update_backlog.hh

```diff
@@ -27,6 +27,12 @@ struct update_backlog {
     size_t current;
     size_t max;
 
+    update_backlog(size_t current, size_t max)
+        : current(current), max(max) {
+            assert(max != 0);
+        }
```

Comment:
Fixed in rebase

Reply:
@piodul do you really feel this is an important thing to worry about?

Reply:
I don't feel this is a thing that I _should_ be worrying about.

I had people pointing out formatting issues to me in my PRs so I thought that people here care about it. After all, we have some rules codified in `coding-style.md`, here I'm just asking to adhere to it. I also personally don't like when code looks inconsistent, it bothers me enough to write a comment.

It's not like it was my only comment and it was delaying the merge of this PR; Wojciech could fix it in 30 seconds while he was addressing other comments. People sometimes leave nitpicks like that.

All in all, I would prefer if there were an automated check for this. We already added automated spell checking, so perhaps we could automate format checks, too?

Reply:
I don't see coding-style.md recommending the specific indentation style you mentioned. The "long lines" thing is just an example of how to indent an if(), I'm not sure how it applies to this case. And to be honest, I don't care. If there isn't a great reason why one minor style variation is better than a second minor style variation, I don't want people's work to be rejected by some automated style nitpicker. And it's a very slippery slope - see what happens in Python in this area (we've had this discussion before), with automated tools taking pride (!) in having **hundreds** of super-nitpicky formatting rules, for no good reason.

Reply:
> I don't see coding-style.md recommending the specific indentation style you mentioned. The "long lines" thing is just an example of how to indent an if(), I'm not sure how it applies to this case.

The rule only mentions breaking long lines, the `if` is just an example and the rule may apply to other cases. I agree though that it is not entirely clear what was the original intention, it is open to interpretation - I think that the rule applies here, you think that it doesn't apply.

With strict rules enforced by a program there would be no need to interpret anything.

> And to be honest, I don't care. If there isn't a great reason why one minor style variation is better than a second minor style variation, I don't want people's work to be rejected by some automated style nitpicker. And it's a very slippery slope - see what happens in Python in this area (we've had this discussion before), with automated tools taking pride (!) in having **hundreds** of super-nitpicky formatting rules, for no good reason.

I must admit that I either don't remember the discussion or didn't participate in it. Could you provide me with a link?

On the other hand, I can say that I only had positive experience enforcing formatting rules in our Rust projects (scylla-rust-driver, cql-stress, scylla-rust-udf, ...) - each one of them enforces a consistent formatting style in CI. It's never a problem as long as you configure your IDE to format the file you are editing when saving it (I think the extension I use configures this automatically). It was never a problem for outside contributors as well.

Of course I am aware that enforcing proper formatting in a project from the start is a different situation than deciding to do it long after it has started - you need people to agree on the rules, potentially modify all the files at once, convince everybody to change their workflow etc. Perhaps it's not worth it at this point. However, if we could hypothetically avoid all the friction resulting from adding the check then I think it would be worth it.